app/src/main/java/com/example/dungeontest/Settings.kt

```diff
@@ -123,22 +225,102 @@ fun SimpleOutlinedTextFieldSample() {
         )
     }
     val textField = FocusRequester()
+    val focusManager = LocalFocusManager.current
+    val textFieldWidth = (maxLength * 8).dp
     OutlinedTextField(
-
-        value = text,
+        value = tokenValue.value,
         onValueChange = {
-            if (it.length <= maxLength) text = it
-            if (it.length == 2) {
-                textField.freeFocus()
-
-            }
+            tokenValue.value = it
+//            Log.d("SimpleOutlinedTextFieldSample", "New value: $text")
         },
 
-        modifier = Modifier.focusRequester(textField),
+        modifier = Modifier.focusRequester(textField).width(textFieldWidth),
         singleLine = true,
         shape = MaterialTheme.shapes.large,
         textStyle = TextStyle(brush = brush),
-        label = { Text("Your OpenAI Key") }
+        label = { Text("Your OpenAI Key") },
+        keyboardActions = KeyboardActions(
+            onDone = {
+                focusManager.clearFocus()
+            }
+        ),
 
     )
-}
\ No newline at end of file
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun OutlinedCardExample(id: Int, title: String, description: String, selectedCard: MutableState<Int>) {
```

Comment:
b4 merge, could you please make the composable name less generic and more clear about its purpose?